src/test/users-query-test.ts

```diff
@@ -108,9 +124,67 @@ describe('users query', () => {
       return 0;
     });
 
-    expect(response.data.data.users).to.deep.equal(users.slice(0, maxUsers));
+    const expectedResponse = {
+      hasNextPage: true,
+      hasPreviousPage: false,
+      totalUsers: qtdUsersToSave,
+      users: users.slice(0, maxUsers),
+    };
+
+    expect(response.data.data.users).to.deep.equal(expectedResponse);
   });
-  it('should not be able to get a user with no token given', async () => {
+
+  it('should get last page of users', async () => {
+    const maxUsers = 30;
+    const skip = 30;
+    const response = await postUsersQuery(token, maxUsers, skip);
+
+    users.sort((a, b) => {
+      if (a.name < b.name) {
+        return -1;
+      }
+      if (a.name > b.name) {
+        return 1;
+      }
+      return 0;
+    });
+
+    const expectedResponse = {
+      hasNextPage: false,
+      hasPreviousPage: true,
+      totalUsers: qtdUsersToSave,
+      users: users.slice(skip, skip + maxUsers),
+    };
+
+    expect(response.data.data.users).to.deep.equal(expectedResponse);
+  });
+
+  it('should get a middle page of users', async () => {
+    const maxUsers = 30;
+    const skip = 10;
+    const response = await postUsersQuery(token, maxUsers, skip);
+
+    users.sort((a, b) => {
+      if (a.name < b.name) {
+        return -1;
+      }
+      if (a.name > b.name) {
+        return 1;
+      }
+      return 0;
+    });
+
+    const expectedResponse = {
+      hasNextPage: true,
+      hasPreviousPage: true,
+      totalUsers: qtdUsersToSave,
+      users: users.slice(skip, skip + maxUsers),
+    };
+
+    expect(response.data.data.users).to.deep.equal(expectedResponse);
+  });
+
+  it('should not be able to get users with no token given', async () => {
```

Comment:
Acho que seria interessante um caso de teste adicional: e se o skip for maior do que o número total de usuários? Nesses casos vc pode retornar um array vazio de usuários.